www/src/components/AddSemesterForm.vue

```diff
@@ -0,0 +1,126 @@
+        console.log(error);
+    }
+}
+
+const translations = {
+    'en': {
+        'Add new semester': 'Add new semester',
+        'Close form': 'Close form',
+        'Semester Type': 'Semester Type',
+        'Winter Semester': 'Winter Semester',
+        'Summer Semester': 'Summer Semester',
+        'Academic Year Start': 'Academic Year Start Year',
+        'Semester Start Date': 'Semester Start Date',
+        'Semester End Date': 'Semester End Date',
+        'Add': 'Add',
+        'Semester with this year and type already exists.': 'Semester with this year and type already exists.',
+    },
+    'pl': {
+        'Add new semester': 'Dodaj nowy semestr',
+        'Close form': 'Zamknij formularz',
+        'Semester Type': 'Rodzaj Semestru',
+        'Winter Semester': 'Semestr Zimowy',
+        'Summer Semester': 'Semestr Letni',
+        'Academic Year Start': 'Rok Rozpoczęcia Roku Akademickiego',
+        'Semester Start Date': 'Data Rozpoczęcia Semestru',
+        'Semester End Date': 'Data zakończenia semestru',
+        'Add': 'Dodaj',
+        'Semester with this year and type already exists.': 'Semestr o podanym roku i rodzaju już istnieje.',
+    },
+};
+
+function translate(text: keyof (typeof translations)['en']): string {
+    return translations[langId][text];
+}
+</script>
+
+<template>
+    <div>
+        <div class="d-flex justify-content-center mb-3">
+            <button type="button" class="btn btn-success" @click="showForm = !showForm">
+                <span v-if="!showForm">{{ translate('Add new semester') }}</span>
+                <span v-else>{{ translate('Close form') }}</span>
+            </button>
+        </div>
+
+        <div
+            v-if="showForm"
+            class="border border-secondary-subtle rounded bg-body-tertiary mx-auto p-3"
+            style="max-width: 600px"
+        >
+            <h2 class="text-center mb-4">{{ translate('Add new semester') }}</h2>
+
+            <form @submit.prevent="submit" class="vstack gap-3 mx-auto" style="max-width: 500px">
+                <div>
+                    <label for="type" class="form-label">{{ translate('Semester Type') }}</label>
+                    <select v-model="type" id="type" class="form-select" required>
+                        <option value="winter">{{ translate('Winter Semester') }}</option>
+                        <option value="summer">{{ translate('Summer Semester') }}</option>
+                    </select>
+                </div>
+
+                <div>
+                    <label for="year" class="form-label">{{ translate('Academic Year Start') }}</label>
+                    <input v-model="year" type="number" id="year" class="form-control" required />
+                </div>
+
+                <div>
+                    <label for="startDate" class="form-label">{{ translate('Semester Start Date') }}</label>
+                    <input v-model="startDate" type="date" id="startDate" class="form-control" required />
```

Comment:
`Date` class cannot be given to v-model. v-model on built-in tags like input will always be a string (except input type="number" which will set model value to type number). You can verify this by placing `{{ typeof startDate }}` somewhere. `startDate` should be of type `string`. Note that vue does not type-check v-model on built-in HTML tags.